src/platform/mbed/PlatformManagerImpl.cpp

```diff
@@ -0,0 +1,176 @@
+#include <new>
+
+#include "platform/internal/CHIPDeviceLayerInternal.h"
+
+#include <platform/PlatformManager.h>
+#include <platform/internal/GenericPlatformManagerImpl.cpp>
+
+namespace chip {
+namespace DeviceLayer {
+
+// TODO: Event and timer processing is not efficient from a memory perspective.
+// Both occupy at least 24 bytes when only 4 bytes is required.
+// An optimized designed could use a separate circular buffer to store events
+// and register a single mbed event in the event queue to process all of them.
+// A similar design can be used for timers.
+
+CHIP_ERROR PlatformManagerImpl::_InitChipStack(void)
+{
+    // Members are initialized by the stack
+    if (!mInitialized)
+    {
+        // reinitialize a new thread if it was terminated earlier
+        mLoopTask.~Thread();
+        new (&mLoopTask) rtos::Thread(osPriorityNormal, CHIP_DEVICE_CONFIG_CHIP_TASK_STACK_SIZE,
+                                      /* memory provided */ nullptr, CHIP_DEVICE_CONFIG_CHIP_TASK_NAME);
+
+        // Reinitialize the EventQueue
+        mQueue.~EventQueue();
+        new (&mQueue) events::EventQueue(event_size * CHIP_DEVICE_CONFIG_MAX_EVENT_QUEUE_SIZE);
+
+        // Reinitialize the Mutex
+        mChipStackMutex.~Mutex();
+        new (&mChipStackMutex) rtos::Mutex();
+    }
+    else
+    {
+        ChipLogError(DeviceLayer, "Trying to reinitialize the stack");
+        return CHIP_ERROR_INCORRECT_STATE;
```

Comment:
Maybe we should follow approach of reusing stuff from CodeUtils.h here as well?
```suggestion
        ExitNow(err = CHIP_ERROR_INCORRECT_STATE);
```

Reply:
The more I code with it the less I'm incline in using it. That's very C++ unfriendly and forces you to put variables at the top and not delay initialization. That is not always an option. 

The worst in my opinion is the inconsistency in term of logging, `ExitNow` does not generate extra log while `Verify...` does.